### PR TITLE
Feature - Modify Exit questionnaire for cancer with last version

### PR DESCRIPTION
**Pull request by Antonior26**

### CODE REVIEW

File: schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+    /**
+    Were potentially actionable variants detected?
+    */
+    CancerActionableVariants actionableVariants;
+
+    }
+
+
+    record CancerVariantLevelQuestions{
```
Comment (gerlv) [inline schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl]: Do we have any field for identifying the variant in cancer cases? I.e. if the user adds their comments for some variant, it looks like there is no way to identify it in the answer that we send to the server

File: schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    E.g. via letter from the clinician (days).
+    */
+    double report_distribution_time;
+
+    /**
+    Total time from result to report.
+    The total time taken from when the analysis of the WGS results started  to a report being received
+    by the patient include any 'waiting' time (days).
+    */
+    double report_distribution_time;
```
Comment (gerlv) [inline schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl]: It has the same attribute name as the previous item

File: schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+    /**
+    Which parts of the WGA were reviewed?
+    */
+    CancerReviewedMDTWGA reviewedInMdtWga;
```
Comment (gerlv) [inline schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl]: `CancerReviewedMDTWGA` isn't in this file, is it `ReviewedParts`?

**Comment by asosinsky** [comment]
Looks fine to me

Isn't it missing long comment for 'Were potentially actionable variants detected?':

Potentially actionable variants are defined as those informing clinical assessment and decision making with regard to diagnostic subtype, prognosis, efficacy of potential therapy, cancer susceptibility or access to a clinical trial. Include those actioned in the past AND those which could be potentially actionable in the future (e.g in the case of a patient under surveillance if they were to develop recurrence of disease).

File: schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl [hunk 4]
```diff
@@ -145,7 +145,7 @@ protocol ExitQuestionnaires {
     Total time taken to review/collate evidence for variants (hours).
     Include all literature review time, consultation with relevant experts etc.
     */
-    double total_time;
+    double total_review_time;
```
Comment (greglever) [inline schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl]: :+1:

File: schemas/IDLs/org.gel.models.report.avro/4.1.0/ReportVersionControl.avdl [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+/**
+This protocol defines the Participants and Pedigree
+*/
+protocol VersionControl {
+
+    record ReportVersionControl {
+
+        /** This is the version for the entire set of data models as referred to the Git release tag
+        */
+        string gitVersionControl = "4.0.0";
```
Comment (priesgo) [inline schemas/IDLs/org.gel.models.report.avro/4.1.0/ReportVersionControl.avdl]: @Antonior26 this should be 4.1.0

File: schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+*/
+protocol ExitQuestionnaires {
+
+    enum CaseSolvedFamily {yes, no, partially, unknown}
+    enum PhenotypesSolved {yes, no, partially, unknown}
+    enum SegregationQuestion {yes, no}
+    enum ReportingQuestion {yes, no, na}
+    enum ConfirmationDecision {yes, no, na}
+    enum ConfirmationOutcome {yes, no, na}
+    enum Actionability {yes, no, not_yet, na}
```
Comment (priesgo) [inline schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl]: What does `not_yet` means as opposed to `no`?

File: schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+    // TODO: refactor with VariantClassification in CommonInterpreted.avdl
+    enum ACMGClassification {pathogenic_variant, likely_pathogenic_variant, variant_of_unknown_clinical_significance, likely_benign_variant,  benign_variant, not_assessed}
+    enum ClinicalUtility {none, change_in_medication, surgical_option, additional_surveillance_for_proband_or_relatives, clinical_trial_eligibility, informs_reproductive_choice, unknown, other}
```
Comment (priesgo) [inline schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl]: What is the difference between `none` and `unknown`?

File: schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+        int variantGroup;
+        array<VariantLevelQuestions> variantLevelQuestions;
+        /**
+        Is evidence for this variant/variant pair sufficient to use it for clinical purposes such as prenatal diagnosis or predictive testing?
+        */
+        Actionability actionability;
+        /**
+        Has the clinical team identified any changes to clinical care which could potentially arise as a result of this variant/variant pair?
+        */
+        array<ClinicalUtility> clinicalUtility;
```
Comment (priesgo) [inline schemas/IDLs/org.gel.models.report.avro/4.1.0/ExitQuestionnaire.avdl]: Having an array of this enum having values of `none` and `unknown` means we need to check on the application layer that no combinations like `none` and `surgical_option` are allowed. Wouldn't it be easier not to have these "null" values?

File: protocols/tests/test_util/test_factories.py [hunk 9]
```diff
@@ -0,0 +1,72 @@
+from unittest import TestCase
+
+import factory
+
+from protocols.ga4gh_3_0_0 import Variant
+from protocols.reports_4_1_0 import CancerExitQuestionnaire
+from protocols.util.factories.ga4gh_factories import GA4GHVariantFactory, CallFactory
+from protocols.util.factories.reports_4_1_0_factoires import CancerExitQuestionnaireFactory
```
Comment (priesgo) [inline protocols/tests/test_util/test_factories.py]: factoires > factories

More to follow.

File: protocols/util/factories/avro_factory.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    expected_schema = protocol.schema
+    basics = ['null', 'boolean', 'string', 'bytes', 'int', 'long', 'float', 'double', 'enum']
+    mock_result = {}
+    for field in expected_schema.fields:
+        schema_type = field.type.type
+        name = field.name
+        if schema_type in basics:
+            basic_types(field, mock_result, name, schema_type)
+        elif schema_type == 'union':
+            basic_types(field.type.schemas[1], mock_result, name, field.type.schemas[1].type)
```
Comment (priesgo) [inline protocols/util/factories/avro_factory.py]: I guess this means we have to always set `union {null, string}` and never union {string, null}. Couldn't we just get the first in the list not being null?

File: protocols/util/factories/avro_factory.py [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    return mock_result
+
+
+def basic_types(field, mock_result, name, schema_type):
+    INT_MIN_VALUE = -(1 << 31)
+    INT_MAX_VALUE = (1 << 31) - 1
+    LONG_MIN_VALUE = -(1 << 63)
+    LONG_MAX_VALUE = (1 << 63) - 1
+    if schema_type == 'null':
+        mock_result[name] = 'null'
```
Comment (priesgo) [inline protocols/util/factories/avro_factory.py]: Shouldn't this be `None` without double quotes?

**Comment by Antonior26** [comment]
{string, union} is not valid

**Pull request merged**